### PR TITLE
Need fix after amphp/sql update

### PR DESCRIPTION
**Pull request by prolic**
resolves https://github.com/amphp/postgres/issues/15

### CODE REVIEW

File: src/PgSqlResultSet.php [hunk 1]
```diff
@@ -243,7 +239,7 @@ public function numRows(): int
     /**
      * @return int Number of fields in each row.
      */
-    public function numFields(): int
+    public function getNumFields(): int
```
Comment (kelunik) [inline src/PgSqlResultSet.php]: Should probably be `getFieldCount()`?

File: src/PgSqlResultSet.php [hunk 2]
```diff
@@ -257,7 +253,7 @@ public function numFields(): int
      */
     public function fieldName(int $fieldNum): string
```
Comment (kelunik) [inline src/PgSqlResultSet.php]: Rename to `getFieldName()`?

File: src/ResultSet.php [hunk 3]
```diff
@@ -11,5 +11,5 @@
      *
      * @return int
      */
-    public function numFields(): int;
+    public function getNumFields(): int;
```
Comment (kelunik) [inline src/ResultSet.php]: Can this be added to `Amp\Sql\ResultSet` or does MySQL not provide this information upfront?

Comment (prolic) [reply src/ResultSet.php]: Amp\Mysql has this: `public function getFields(): Promise`

But it's written with Amp\Socket, so we can probably refactor it.

**Pull request merged**